metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -8268,29 +8279,76 @@ public NotificationEventResponse getNextNotification(NotificationEventRequest rq
     }
   }
 
+  private void lockForUpdate() throws MetaException {
+    String selectQuery = "select NEXT_EVENT_ID from NOTIFICATION_SEQUENCE";
+    String selectForUpdateQuery = sqlGenerator.addForUpdateClause(selectQuery);
+    new RetryingExecutor(3, () -> {
```

Comment:
self-review: do i need to have a hiveConf setting for this ?

Reply:
This would be quite useful.  Also, it makes sense to use something like exponential backoff. 
See https://github.com/apache/sentry/blob/master/sentry-provider/sentry-provider-db/src/main/java/org/apache/sentry/provider/db/service/persistent/TransactionManager.java - in particular, ExponentialBackoff class.